types.go

```diff
@@ -156,5 +159,17 @@ type Error struct {
 
 // Error returns a textual description of the test error.
 func (err Error) Error() string {
+	if strings.TrimSpace(err.Body) != "" {
+		return err.Body
+	}
+
+	if strings.TrimSpace(err.Message) != "" {
+		return err.Message
+	}
+
+	if strings.TrimSpace(err.Type) != "" {
+		return err.Type
+	}
+
 	return err.Body
```

Comment:
At the very end, do you want to `return err.Type` with no check? The value of `err.Body` is **guaranteed** to be blank (or at best all whitespace characters) by this point.

Reply:
Well, my thinking was "try all the fields, if everything is still empty then return Body". In this case we can maintain backward compatibility, since so far only `Body` was used. I'm not sure if we need to change that behaviour. WDYT @joshdk ?

Reply:
Since `err.Body` is both the first thing (preferably) returned, **and** the fallback, it will be blank in the latter case. Concretely, I was thinking of something like:

```golang
switch {
    case strings.TrimSpace(err.Body) != "":
        return err.Body

    case strings.TrimSpace(err.Message) != "":
        return err.Message

    default:
        return err.Type
}
```

There isn't really a way to maintain data back-compat in this case (as the function can _possibly_ return more than just that _single_ value now), but I would be fine with the above refactor 🙂

Reply:
all right, sounds good to me 👍